### PR TITLE
Torrentday: Switched back to the new rest api.

### PR DESCRIPTION
**Pull request by p0psicles**
- [x] PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read the [contribution guide](https://github.com/pymedusa/Medusa/blob/master/.github/CONTRIBUTING.md)

With special tnx to: @kayl669

### CODE REVIEW

File: medusa/providers/torrent/json/torrentday.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+                if mode != 'RSS':
+                    log.debug('Search string: {search}',
+                              {'search': search_string})
+
+                search_string = '+'.join(search_string.split())
+
+                params = dict({'q': search_string}, **self.categories[mode])
+
+                if self.freeleech:
+                    params.update({'free': 'on'})
```
Comment (p0psicles) [inline medusa/providers/torrent/json/torrentday.py]: I'm not sure if this is still working.
@duramato can you test this for me?

**Comment by p0psicles** [comment]
@duramato can you test downloading some torrents for me?

File: medusa/providers/torrent/json/torrentday.py (hunk 1 above)
Comment (p0psicles) [inline medusa/providers/torrent/json/torrentday.py]: I'm not sure if this still has any effect?

Comment (fernandog) [reply medusa/providers/torrent/json/torrentday.py]: afaik its to only list free leech torrents. @duramato confirm?

Comment (p0psicles) [reply medusa/providers/torrent/json/torrentday.py]: Yeah but is it available through the new api?

**Comment by p0psicles** [comment]
reminder to myself. Remove the freeleech param. And filter using `"download-multiplier":0`

**Comment by fernandog** [comment]
@p0psicles where is the API documentation? I couldn't find

**Comment by p0psicles** [comment]
Don't think there is any

File: medusa/providers/torrent/json/torrentday.py [hunk 2]
```diff
@@ -138,6 +135,10 @@ def parse(self, data, mode):
                                   " minimum seeders: {0}. Seeders: {1}", title, seeders)
                     continue
 
+                # Check if this is a freeleech torrent and if we've configured to only allow freeleech.
+                if self.freeleech and row['download-multiplier'] != 0:
+                    continue
```
Comment (fernandog) [inline medusa/providers/torrent/json/torrentday.py]: @p0psicles why not discard freeleech earlier before setting `title` ?  `title = re.sub(...` - line 120

**Pull request merged**